lib/SimpleSAML/Memcache.php

```diff
@@ -154,7 +162,12 @@ public static function set($key, $value, $expire = null)
 
         // store this object to all groups of memcache servers
         foreach (self::getMemcacheServers() as $server) {
-            $server->set($key, $savedInfoSerialized, 0, $expire);
+            if (self::$extension == 'memcached') {
```

Comment:
Tim already mentioned this, but just for the record: strict comparisons are always preferred, even if we are completely sure that the types are not going to be an issue, it is in general a good practice to be on the safe side.